price_feeds_poller.py

```diff
@@ -376,7 +389,7 @@ def log_loop(
                   print(f"{caption} .. no routed update detected on contract {contract.address}")
                 continue
 
-              elif elapsed_secs >= pf["heartbeat"] - network_witnet_resolution_secs * (1 - pf["isRouted"]):
+              elif elapsed_secs >= pf["heartbeat"] -(network_witnet_resolution_secs + network_evm_finalization_secs) * (1 - pf["isRouted"]):
```

Comment:
```suggestion
              elif elapsed_secs >= pf["heartbeat"] - (0 if pf["isRouted"] else (network_witnet_resolution_secs + network_evm_finalization_secs)):
```

You could avoid this multiplication if you want, whatever looks easier to read.